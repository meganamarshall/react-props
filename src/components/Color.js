import React from 'react';
import PropTypes from 'prop-types';

const hexFromColor = color => {
  return `${color.toString(16).padStart(2, 0)}`.toUpperCase();
};

const hexFromRGB = (rgb) => {
  return `#${hexFromColor(rgb.red)}${hexFromColor(rgb.green)}${hexFromColor(rgb.blue)}`;
};


function Color({ name, rgb }) {
  return (
    <dl>
      <dt>Name:</dt>
      <dd>{name}</dd>

      <dt>Hex:</dt>
      <dd>{hexFromRGB(rgb)}</dd>

      <dt>RGB:</dt>
      <dd>
        <p>Red - {rgb.red}</p>
        <p>Green - {rgb.green}</p>
        <p>Blue - {rgb.blue}</p>
      </dd>
    </dl>
  );
}

Color.propTypes = {
  name: PropTypes.string.isRequired,
  rgb: {
    red: PropTypes.string.isRequired,
    blue: PropTypes.string.isRequired,
    green: PropTypes.string.isRequired
  }
};

export default Color;
