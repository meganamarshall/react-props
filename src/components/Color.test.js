import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color component', () => {
  it('matches the snapshot', () => {
    const testColorName = 'blue';
    const rgb = {
      red: 0,
      green: 0,
      blue: 255
    };
    const wrapper = shallow(<Color name={testColorName} rgb={rgb} />);
    expect(wrapper).toMatchSnapshot();

  });
});
