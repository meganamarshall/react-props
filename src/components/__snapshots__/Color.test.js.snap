// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color component matches the snapshot 1`] = `
<dl>
  <dt>
    Name:
  </dt>
  <dd>
    blue
  </dd>
  <dt>
    Hex:
  </dt>
  <dd>
    #0000FF
  </dd>
  <dt>
    RGB:
  </dt>
  <dd>
    <p>
      Red - 
      0
    </p>
    <p>
      Green - 
      0
    </p>
    <p>
      Blue - 
      255
    </p>
  </dd>
</dl>
`;
